sycl/include/CL/sycl/detail/image_impl.hpp

```diff
@@ -8,150 +8,318 @@
+  range<dimensions> get_range() const { return Range; }
+
+  // Available only when: dimensions > 1.
+  // Return a range object representing the pitch of the image in bytes.
+  template <bool B = (dimensions == 2)>
+  typename std::enable_if<B, range<1>>::type get_pitch() const {
+    range<1> temp = range<1>(RowPitch);
+    return temp;
+  }
 
-  //image_impl(cl_mem clMemObject, const context &syclContext,
-             //event availableEvent = {});
+  template <bool B = (dimensions == 3)>
+  typename std::enable_if<B, range<2>>::type get_pitch() const {
+    range<2> temp = range<2>(RowPitch, SlicePitch);
+    return temp;
+  }
 
-  /* -- property interface members -- */
+  // Returns the size of the image storage in bytes
+  size_t get_size() const { return MSizeInBytes; }
 
-  range<dimensions> get_range() const { assert(!"Not implemented"); }
+  // Returns the total number of elements in the image
+  size_t get_count() const { return Range.size(); }
 
-  /* Available only when: dimensions > 1 */
-  range<dimensions - 1> get_pitch() const { assert(!"Not implemented"); }
+  // Returns the allocator provided to the image
+  AllocatorT get_allocator() const { return MAllocator; }
 
-  size_t get_size() const { assert(!"Not implemented"); return 0;}
+  template <typename propertyT> bool has_property() const {
+    return MProps.has_property<propertyT>();
+  }
 
-  size_t get_count() const { assert(!"Not implemented"); return 0; }
+  template <typename propertyT> propertyT get_property() const {
+    return MProps.get_property<propertyT>();
+  }
 
-  AllocatorT get_allocator() const { assert(!"Not implemented"); }
+  void *allocateHostMem() override {
+    assert(!"Not Implemented");
+    return nullptr;
+    // TODO: Implement this function.
+    // Implementation of the pure virtual function.
+  }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::image>
-  get_access(handler &commandGroupHandler) {
-    assert(!"Not implemented");
+  void *allocateMem(ContextImplPtr Context, bool InitFromUserData,
+                    cnri_event &OutEventToWait) override {
+    assert(!"Not Implemented");
+    return nullptr;
+    // TODO: Implement this function.
+    // Implementation of the pure virtual function.
   }
 
-  template <typename dataT, access::mode accessMode>
-  accessor<dataT, dimensions, accessMode, access::target::host_image>
-  get_access() {
-    assert(!"Not implemented");
+  MemObjType getType() const override { return MemObjType::IMAGE; }
+
+  void releaseHostMem(void *Ptr) override {
+    assert(!"Not Implemented");
+    return;
+    // TODO: Implement this function.
+    // Implementation of the pure virtual function.
   }
 
-  // template <typename Destination = std::nullptr_t>
-  // void set_final_data(Destination finalData = std::nullptr);
+  void releaseMem(ContextImplPtr Context, void *MemAllocation) override {
+    assert(!"Not Implemented");
```

Comment:
Please, replace all assertions like this with exceptions.

Reply:
Changed.